src/analyses/cfg_dominators.h

```diff
@@ -142,96 +293,201 @@ void cfg_dominators_templatet<P, T, post_dom>::initialise(P &program)
   cfg(program);
 }
 
+template <class P, class T, bool post_dom>
+void cfg_dominators_templatet<P, T, post_dom>::assign_postordering(
+  std::size_t start_node_index)
+{
+  struct stack_entryt
+  {
+    // cfg index, not the post-order index we're assigning
+    std::size_t node_index;
+    typename cfgt::nodet::edgest::const_iterator child_iterator;
+    typename cfgt::nodet::edgest::const_iterator child_end;
+  };
+
+  std::size_t next_postorder_index = 0;
+  std::vector<stack_entryt> stack;
+
+  auto place_node_on_stack_if_not_visited = [this,
+                                             &stack](std::size_t node_index) {
+    // If we've already visited this node, don't re-calculate.
+    auto &node = cfg[node_index];
+    if(node.postorder_index != nodet::NODE_NOT_VISITED)
+      return;
+
+    // Otherwise set that we've processed it, and put children on the stack.
```

Comment:
We don't put the children, we put the individual node (along with initialising that node's child iterators), as the method name suggests

Reply:
It puts `node_successors` on the stack immediately under this for processing on the next iteration, which is what it's talking about. Though successors might be a better word than children. It's mostly superfluous anyway as it's just describing the next two lines anyway.